### PR TITLE
[SPARK-37678][PYTHON] Fix _cleanup_and_return signature

### PR DESCRIPTION
**Pull request by zero323**
### What changes were proposed in this pull request?

This PR fixes return type annotation for  `pandas.groupby.SeriesGroupBy._cleanup_and_return`.

### Why are the changes needed?

Current annotation is incorrect (mixes pandas an pyspark.pandas types).

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @ueshin FYI

File: python/pyspark/pandas/groupby.py [hunk 1]
```diff
@@ -2994,7 +2994,7 @@ def _apply_series_op(
         else:
             return psser.copy()
 
-    def _cleanup_and_return(self, pdf: pd.DataFrame) -> Series:
+    def _cleanup_and_return(self, pdf: DataFrame) -> Series:
```
Comment (ueshin) [inline python/pyspark/pandas/groupby.py:2997]: nit: shall we use `psdf` for the variable name of `DataFrame`?

**Review by ueshin — APPROVED** [review]
Otherwise, LGTM.

**Comment by zero323** [comment]
Jenkins, retest this please.

**Comment by ueshin** [comment]
Thanks! merging to master/3.2

**Pull request closed without merging**